### PR TITLE
perf(master): Optimize MP split algorithm

### LINKED ISSUE #2511: [Enhancement]: Optimize MetaPartition Split Algorithm
### Contact Details

tangdeyi@oppo.com

### Is there an existing issue for this?

- [X] I have searched all the existing issues

### What would you like to be added?

Optimize MetaPartition Split Algorithm

### Why is this needed?

The meta partition split algorithm should be optimized for the following reason：
1. Avoiding too many inodes managed by the last meta partition.
2. Improving the write availability of volume. 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tangdeyi**
**Which issue this PR fixes**:

fixes #2511

### CODE REVIEW

**Review by baijiaruo — APPROVED** [review]

File: master/vol.go [hunk 1]
```diff
@@ -509,18 +520,35 @@ func (vol *Vol) checkMetaPartitions(c *Cluster) {
...
 		return
 	}
-	vol.mpsLock.RUnlock()
+	// Any of the following conditions will trigger max mp split
+	// 1. The memory of the metanode which max mp belongs to reaches the threshold
+	// 2. The number of inodes managed by max mp reaches the threshold(0.75)
+	// 3. The number of RW mp is less than 2
+	maxMPInodeUsedRatio := float64(maxMP.MaxInodeID-maxMP.Start) / float64(metaPartitionInodeStep)
+	RWMPNum := vol.getRWMetaPartitionNum()
+	if maxMP.memUsedReachThreshold(c.Name, vol.Name) || RWMPNum < 2 || maxMPInodeUsedRatio > 0.75 {
```
Comment (baijiaruo) [inline master/vol.go]: Don't use magic numbers

**Review by Victor1319 — APPROVED** [review]
LGTM

**Review by baijiaruo — APPROVED** [review]
LGTM

**Review by Victor1319 — APPROVED** [review]
LGTM

**Pull request merged**